TheMovieDatabase/ViewControllers/Favorites/View/FavoritesViewController.swift

```diff
@@ -146,14 +140,14 @@ extension FavoritesViewController: UITableViewDataSource {
             return
         }
         if editingStyle == .delete {
-            if Locksmith.getSessionId() != nil {
+            if authorizationService.getSessionId() != nil {
```

Comment:
The same here, after notification implementation I expect to see there only `profileService.setFavorite(to: Bool, movieId: Int, completion: (Result<Void, Error>))` and notice that you shouldn't do there anything concerned tableView, it would be handled in subscription to notifications

And hint for subscription function:
```
service.subscribeToFavoriteMovies(onChange: (Void) -> ())
```